kusto/kcsb.go

```diff
@@ -0,0 +1,354 @@
+	"application key": applicationKey, "applicationkey": applicationKey, "appkey": applicationKey,
+	"application certificate": applicationCertificate, "applicationcertificate": applicationCertificate,
+	"application certificate thumbprint": applicationCertificateThumbprint, "applicationcertificatethumbprint": applicationCertificateThumbprint,
+	"sendcertificatechain": sendCertificateChain, "send certificate chain": sendCertificateChain,
+	"authority id": authorityId, "authorityid": authorityId, "authority": authorityId, "tenantid": authorityId, "tenant": authorityId, "tid": authorityId,
+	"application token": applicationToken, "applicationtoken": applicationToken, "apptoken": applicationToken,
+	"user token": userToken, "usertoken": userToken, "usrtoken": userToken,
+	"msi_auth":               msiAuth,
+	"managedserviceidentity": managedServiceIdentity, "managed service identity": managedServiceIdentity,
+	"interactive login": interactiveLogin, "interactivelogin": interactiveLogin,
+	"az cli": azCli, "azcli": azCli,
+	"domain hint": domainHint, "domainhint": domainHint,
+}
+
+func requireNonEmpty(key string, value string) {
+	if isEmpty(value) {
+		panic(fmt.Sprintf("Error: %s cannot be null", key))
+	}
+}
+
+func assignValue(kcsb *connectionStringBuilder, rawKey string, value string) {
+	rawKey = strings.ToLower(strings.Trim(rawKey, " "))
+	parsedKey := csMapping[rawKey]
+	switch parsedKey {
+	case dataSource:
+		kcsb.dataSource = value
+	case aadUserId:
+		kcsb.aadUserID = value
+	case password:
+		kcsb.password = value
+	case applicationClientId:
+		kcsb.applicationClientId = value
+	case applicationKey:
+		kcsb.applicationKey = value
+	case applicationCertificate:
+		kcsb.applicationCertificate = value
+	case applicationCertificateThumbprint:
+		kcsb.applicationCertificateThumbprint = value
+	case sendCertificateChain:
+		bval, _ := strconv.ParseBool(value)
+		kcsb.sendCertificateChain = bval
+	case authorityId:
+		kcsb.authorityId = value
+	case applicationToken:
+		kcsb.applicationToken = value
+	case userToken:
+		kcsb.userToken = value
+	case msiAuth:
+		bval, _ := strconv.ParseBool(value)
+		kcsb.msiAuthentication = bval
+	case managedServiceIdentity:
+		kcsb.managedServiceIdentity = value
+	case azCli:
+		bval, _ := strconv.ParseBool(value)
+		kcsb.azcli = bval
+	case interactiveLogin:
+		bval, _ := strconv.ParseBool(value)
+		kcsb.interactiveLogin = bval
+	case domainHint:
+		kcsb.redirectURL = value
+	}
+}
+
+/*
+   Creates new KustoConnectionStringBuilder.
+   Params takes kusto connection string connStr: string.  Kusto connection string should be of the format:
+   https://<clusterName>.kusto.windows.net;AAD User ID="user@microsoft.com";Password=P@ssWord
+   For more information please look at:
+   https://docs.microsoft.com/azure/data-explorer/kusto/api/connection-strings/kusto
+*/
+func GetConnectionStringBuilder(connStr string) connectionStringBuilder {
```

Comment:
Why not use New()

Reply:
`Kusto.New()` is used for obtaining the `Client`. 
`Kusto.GetConnectionStringBuilder()` becomes better way. 
Either we go for whole new package name `Connectionstring`. So the `ConnectionString.New()` can be used..?

Reply:
@AsafMah WDYT?
New package or change the API name to **New**ConnectionStringBuilder?